lib/util/string_conv/string_conv.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef STRING_CONV_H__
+#define STRING_CONV_H__
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Convert string to long param.
+ *
+ * @note On failure the passed value reference will not be altered.
+ *
+ * @param str Input string
+ * @param val Converted value
+ *
+ * @return 0 on success, negative error code on failure
+ */
+int string_conv_str2long(char *str, int *val);
+
+/**
+ * @brief Convert string to unsigned long param.
+ *
+ * @note On failure the passed value reference will not be altered.
+ *
+ * @param str Input string
+ * @param val Converted value
+ *
+ * @return 0 on success, negative error code on failure
+ */
+int string_conv_str2ulong(char *str, uint64_t *val);
+
+/**
+ * @brief Convert string to double param.
+ *
+ * @note On failure the passed value reference will not be altered.
+ *
+ * @param str Input string
+ * @param val Converted value
+ *
+ * @return 0 on success, negative error code on failure
+ */
+int string_conv_str2dbl(const char *str, double *val);
```

Comment:
What is proper string for conversion? Is `  -10.20    ` acceptable?

Reply:
"-10.20" is fine, "-10.20 " is not.

Reply:
So the user would have to trim the string to the acceptable value anyway?
So basically a user would have to recognize whether a string is acceptable then the function will check whether string is acceptable, and then it will make dbl of it.
Wouldn't it be better to just allow it to skip white chars that immediately precede or follow the number and return, similarly to `strtol` pointer to last unprocessed character and let user decide whether it is acceptable that not all characters in buffer were consumable?

Reply:
You have a point 👍 I will add func to ignore heading/trailing whitespace.